lib/nostrum/api.ex

```diff
@@ -741,15 +746,31 @@ defmodule Nostrum.Api do
   def bulk_delete_messages(channel_id, messages, filter \\ true)
 
   def bulk_delete_messages(channel_id, messages, false),
-    do: request(:post, Constants.channel_bulk_delete(channel_id), %{messages: messages})
+    do: send_chunked_delete(messages, channel_id)
 
   def bulk_delete_messages(channel_id, messages, true) do
-    filtered_messages =
-      Enum.filter(messages, fn message_id ->
-        (message_id >>> 22) + 1_420_070_400_000 > Util.now() - 14 * 24 * 60 * 60 * 1000
-      end)
+    messages
+    |> Stream.filter(fn message_id ->
+      (message_id >>> 22) + 1_420_070_400_000 > Util.now() - 14 * 24 * 60 * 60 * 1000
```

Comment:
Now that #90 is implemented, we should create a snowflake here by taking the current date and pushing it back two weeks and comparing against that.